Use Material-UI

nickserv opened the pull request:
Fixes #13.

nickserv committed:
Set up material-ui

nickserv committed:
Replace react-bootstrap with material-ui components

nickserv committed:
Use FontIcon instead of react-fa

nickserv committed:
Minor import fix

nickserv committed:
Use material-ui/svg-icons instead of FontIcon with Font Awesome

nickserv committed:
Remove useless container class

nickserv committed:
Merge branch 'master' into material-ui

nickmccurdy commented on public/index.html:
```diff
@@ -27,16 +27,16 @@
-    -->
+         To begin the development, run `npm start` or `yarn start`.
+         To create a production bundle, use `npm run build` or `yarn build`.
+       -->
```
Undo whitespace changes

nickmccurdy commented on src/components/Options/Custom.js:
```diff
@@ -1,16 +1,14 @@
+import { TextField } from 'material-ui'
+import { ImageEdit } from 'material-ui/svg-icons'
```
Optimize imports

nickmccurdy commented on src/components/Options/Diceware.js:
```diff
@@ -1,13 +1,15 @@
+import { TextField } from 'material-ui'
+import { ImageEdit } from 'material-ui/svg-icons'
```
Optimize imports

nickmccurdy commented on src/components/Options/Diceware.js:
```diff
@@ -1,13 +1,15 @@
 import { dicewareWords } from '../../reducers/possibleItemsSelector'
 
-export default function Diceware() {
+export default function DicewareOptions() {
```
Fix name

nickmccurdy commented on src/components/Options/Generic.js:
```diff
@@ -1,9 +1,9 @@
 import startCase from 'lodash/startCase'
+import { Checkbox } from 'material-ui'
```
Optimize imports

nickmccurdy commented on src/components/Options/index.js:
```diff
@@ -1,11 +1,9 @@
 import startCase from 'lodash/startCase'
+import { Card, CardText, Tab, Tabs } from 'material-ui'
+import { ActionBook, ImageEdit, NavigationCheck } from 'material-ui/svg-icons'
```
Optimize imports

nickmccurdy commented on src/components/PasswordEntropy.js:
```diff
@@ -1,32 +1,27 @@
+import { TextField } from 'material-ui'
+import { ActionCompareArrows, ActionInfo, ActionLock } from 'material-ui/svg-icons'
```
Optimize imports

nickmccurdy commented on src/components/PasswordEntropy.js:
```diff
@@ -1,32 +1,27 @@
+      <h1>
+        <ActionLock /> Password Entropy
         <br />
         <small>A simple entropy calculator for evaluating password security.</small>
```
Normal sized text to save space on mobile

nickmccurdy commented on src/components/PossiblePasswords.js:
```diff
@@ -1,34 +1,37 @@
+    <Card>
+      <CardText>
+        <h3><ActionList /> Entropy</h3>
+        {entropyBits.toFixed(2)} bits ({strength})
```
Better styling

nickmccurdy commented on src/components/PossiblePasswords.test.js:
```diff
@@ -1,11 +1,12 @@
 import { shallow } from 'enzyme'
+import { colors } from 'material-ui/styles'
```
Optimize imports

nickmccurdy commented on src/index.js:
```diff
@@ -1,16 +1,20 @@
+import { MuiThemeProvider } from 'material-ui'
```
Optimize imports

nickmccurdy commented on src/index.js:
```diff
@@ -1,16 +1,20 @@
 import { createStore } from 'redux'
 import { devToolsEnhancer } from 'redux-devtools-extension'
-import 'bootstrap/dist/css/bootstrap.css'
+import 'roboto-fontface/css/roboto/roboto-fontface.css'
```
Try root import

nickmccurdy commented on src/index.js:
```diff
@@ -1,16 +1,20 @@
-    <PasswordEntropy />
+    <MuiThemeProvider>
+      <PasswordEntropy />
+    </MuiThemeProvider>
```
Look into Component compose functions

nickmccurdy commented on src/reducers/possibleItemsSelector.js:
```diff
@@ -1,3 +1,4 @@
+import { colors } from 'material-ui/styles'
```
Optimize imports

nickserv committed:
Fix index.html comment indentation

nickserv committed:
Optimize imports

nickserv committed:
Move description out of h1

nickserv committed:
Use Material Design lock icon as favicon

nickserv committed:
Set theme color to cyan500

nickserv committed:
Add missing paragraph tags

nickserv committed:
Update snapshots

nickserv committed:
Use inline styles

nickserv committed:
Fix package.json formatting

nickserv committed:
Remove brittle render callback tests

nickserv committed:
Update to material-ui 1 beta

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Use react-test-renderer/shallow directly instead of Enzyme

nickserv committed:
Switch from roboto-fontface to typeface-roboto

nickserv committed:
Compose scripts with npm-run-all

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Always use caret version ranges

nickserv committed:
Update dependencies

nickserv committed:
Use native Object.entries instead of shim

nickserv committed:
Capitalize Generic Options and drop lodash dependency

nickserv committed:
Revert "Capitalize Generic Options and drop lodash dependency"

This reverts commit a15f50be37b5e44e4d1a4821b7aedef49b436ed0.

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Use CssBaseline

nickserv committed:
Use Fragment

nickserv committed:
Run Prettier

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Replace span with Fragment

nickserv committed:
Prune dependencies

nickserv committed:
Clean up imports

nickserv committed:
Remove icons

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Use Typography component

nickserv committed:
Format

nickserv committed:
Update snapshots

nickserv committed:
Remove Diceware component

The read only Possible Items field should be hidden to prevent confusion, since it's an implementation detail.

nickserv committed:
Remove required and use autofocus

nickserv committed:
Swap results order for readability

nickserv committed:
Move Results title into card

nickserv committed:
Move all options into a card

nickserv committed:
Remove form tag

nickserv committed:
Use two grid columns

nickserv committed:
Use more consistent line spacing

nickserv committed:
Format

nickserv committed:
Rename Length to Number of Items

nickserv committed:
Rename letters

nickserv committed:
Use tabs in an app bar for the whole page

nickserv committed:
Simplify grid syntax

nickserv committed:
Rename components

nickserv committed:
Merge length field into options

nickserv committed:
Move components back into src

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Merge branch 'master' into material-ui

nickserv committed:
Fix theme colors

nickserv committed:
Update Material UI

nickserv committed:
Update snapshots

nickserv committed:
Update dependencies

nickserv committed:
Merge branch 'master' into material-ui
